### PR TITLE
App-1342-Incorrect status code on Integration Bridge Authorization API

### PR DESCRIPTION
**Pull request by Hamitay-Daitan**
Added status code on JiraOAuth1Exception and changed exception handling on JiraWebHookIntegration

### CODE REVIEW

File: src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java [hunk 1]
```diff
@@ -171,13 +172,13 @@ public void authorize(AuthorizationPayload authorizationPayload) throws Authoriz
 
     if (StringUtils.isBlank(temporaryToken) || StringUtils.isBlank(verificationCode)) {
       throw new JiraOAuth1Exception(MSG.getMessage(MSG_INSUFFICIENT_PARAMS),
-          MSG.getMessage(MSG_INSUFFICIENT_PARAMS_SOLUTION));
+          HttpStatus.BAD_REQUEST.value(), MSG.getMessage(MSG_INSUFFICIENT_PARAMS_SOLUTION));
```
Comment (campidelli) [inline src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java]: I don't think we should handle HTTP codes in this layer. This layer should be independent of protocols. It's better to extend JiraOAuth1Exception and create a meaningful Exception to express what you want here.

File: src/main/java/org/symphonyoss/integration/jira/authorization/oauth/v1/JiraOAuth1Exception.java [hunk 2]
```diff
@@ -17,5 +16,10 @@ public JiraOAuth1Exception(String message, Throwable cause, String... solutions)
   public JiraOAuth1Exception(String message, String... solutions) {
     super(message, solutions);
   }
+
+  public JiraOAuth1Exception(String message, int code,
```
Comment (campidelli) [inline src/main/java/org/symphonyoss/integration/jira/authorization/oauth/v1/JiraOAuth1Exception.java]: Http code here? I prefer to use them only in the RemoteApiException.

File: src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java [hunk 3]
```diff
@@ -171,13 +172,13 @@ public void authorize(AuthorizationPayload authorizationPayload) throws Authoriz
...
     }
 
     IntegrationSettings settings = getSettings();
     if (settings == null) {
       throw new JiraOAuth1Exception(MSG.getMessage(MSG_NO_INTEGRATION_FOUND),
-          MSG.getMessage(MSG_NO_INTEGRATION_FOUND_SOLUTION));
+          HttpStatus.NOT_FOUND.value(), MSG.getMessage(MSG_NO_INTEGRATION_FOUND_SOLUTION));
```
Comment (campidelli) [inline src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java]: Same comment as above.

**Review by campidelli — CHANGES_REQUESTED** [review]

File: src/main/java/org/symphonyoss/integration/jira/authorization/oauth/v1/exception/JiraOAuth1IntegrationNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package org.symphonyoss.integration.jira.authorization.oauth.v1.exception;
+
+import org.symphonyoss.integration.authorization.oauth.v1.OAuth1IntegrationNotFoundException;
+
+/**
+ * Created by hamitay on 17/10/17.
+ */
+public class JiraOAuth1IntegrationNotFoundException extends OAuth1IntegrationNotFoundException {
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/jira/authorization/oauth/v1/exception/JiraOAuth1IntegrationNotFoundException.java]: I think this class is not useful for a while. Please remove this one.

File: src/main/java/org/symphonyoss/integration/jira/authorization/oauth/v1/exception/JiraOAuth1MissingParametersException.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package org.symphonyoss.integration.jira.authorization.oauth.v1.exception;
+
+import org.symphonyoss.integration.authorization.oauth.v1.OAuth1MissingParametersException;
+
+/**
+ * Created by hamitay on 17/10/17.
+ */
+public class JiraOAuth1MissingParametersException extends OAuth1MissingParametersException {
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/jira/authorization/oauth/v1/exception/JiraOAuth1MissingParametersException.java]: Same comment above.

File: src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java [hunk 6]
```diff
@@ -170,13 +174,13 @@ public void authorize(AuthorizationPayload authorizationPayload) throws Authoriz
     String verificationCode = authorizationPayload.getParameters().get(OAUTH_VERIFIER);
 
     if (StringUtils.isBlank(temporaryToken) || StringUtils.isBlank(verificationCode)) {
-      throw new JiraOAuth1Exception(MSG.getMessage(MSG_INSUFFICIENT_PARAMS),
+      throw new JiraOAuth1MissingParametersException(MSG.getMessage(MSG_INSUFFICIENT_PARAMS),
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java]: You should throw OAuth1MissingParametersException directly.

File: src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java [hunk 7]
```diff
@@ -170,13 +174,13 @@ public void authorize(AuthorizationPayload authorizationPayload) throws Authoriz
...
           MSG.getMessage(MSG_INSUFFICIENT_PARAMS_SOLUTION));
     }
 
     IntegrationSettings settings = getSettings();
     if (settings == null) {
-      throw new JiraOAuth1Exception(MSG.getMessage(MSG_NO_INTEGRATION_FOUND),
+      throw new JiraOAuth1IntegrationNotFoundException(MSG.getMessage(MSG_NO_INTEGRATION_FOUND),
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/jira/webhook/JiraWebHookIntegration.java]: You should throw OAuth1IntegrationNotFoundException directly.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**